examples/experimental/dagster-airlift/dagster_airlift/core/defs_builders.py

```diff
@@ -0,0 +1,42 @@
+from typing import Sequence, Union
+
+from dagster import AssetsDefinition, AssetSpec, multi_asset
+from dagster._core.definitions.asset_key import CoercibleToAssetKey
+from dagster._core.definitions.definitions_class import Definitions
+from typing_extensions import TypeAlias
+
+from .utils import DAG_ID_TAG, TASK_ID_TAG
+
+CoercibleToAssetSpec: TypeAlias = Union[AssetSpec, CoercibleToAssetKey]
+
+
+def defs_from_task(
+    *, task_id: str, dag_id: str, assets: Sequence[CoercibleToAssetSpec]
+) -> Definitions:
+    """Construct a Dagster :py:class:`Definitions` object from a provided set of assets,
+    with a mapping to which airflow task produces those assets.
+    """
+    asset_specs = [
+        asset if isinstance(asset, AssetSpec) else AssetSpec(key=asset) for asset in assets
+    ]
+
+    @multi_asset(
+        op_tags={DAG_ID_TAG: dag_id, TASK_ID_TAG: task_id},
+        specs=asset_specs,
+        name=f"{dag_id}__{task_id}",
+        compute_kind="airflow",
+    )
+    def task_asset():
+        pass
+
+    return Definitions(assets=[task_asset])
```

Comment:
@benpankow let's note this as the first place we should refactor once we ship `kinds` attached to assets rather than at the op level